dojo/tools/whitehat_sentinel/parser.py

```diff
@@ -0,0 +1,244 @@
+                                               fragment=fragment)
+                                      )
+            except Exception:
+                pass
+
+        return endpoints_list
+
+    def _convert_whitehat_sentinel_vuln_to_dojo_finding(self, whitehat_sentinel_vuln, test):
+        """
+        Converts a WhiteHat Sentinel vuln to a DefectDojo finding
+
+        Args:
+            whitehat_sentinel_vuln: The vuln dictionary from WhiteHat Sentinel vuln API
+            test: The test ID that the DefectDojo finding should be associated with
+        Returns: A DefectDojo Finding object
+        """
+
+        # Out of scope is considered Active because the issue is valid, just not for the asset in question. If the
+        # issue belongs to another Product, it should be filtered before importing.
+        active = whitehat_sentinel_vuln.get('status') in ('open', 'out of scope')
+        date_created = whitehat_sentinel_vuln['found'].split('T')[0]
+        mitigated_ts = whitehat_sentinel_vuln.get('closed'.split('T')[0], None)
+        cwe = self._parse_cwe_from_tags(whitehat_sentinel_vuln['attack_vectors'][0]['scanner_tags'])
+        description_ref = self._parse_description(whitehat_sentinel_vuln['description'])
+        description = description_ref['description']
+        references = description_ref['reference_link']
+        steps = whitehat_sentinel_vuln['description'].get('description_prepend', '')
+        solution = self._parse_solution(whitehat_sentinel_vuln['solution'])
+        risk_id = whitehat_sentinel_vuln.get('custom_risk') if whitehat_sentinel_vuln.get(
+            'custom_risk') else whitehat_sentinel_vuln.get('risk')
+        severity = self._convert_whitehat_severity_id_to_dojo_severity(risk_id)
+        false_positive = whitehat_sentinel_vuln.get('status') == 'invalid'
+
+        finding = Finding(title=whitehat_sentinel_vuln['class'],
+                          test=test,
+                          cwe=cwe,
+                          active=active,
+                          verified=True,
+                          description=description,
+                          steps_to_reproduce=steps,
+                          mitigation=solution,
+                          references=references,
+                          severity=severity,
+                          numerical_severity=Finding.get_numerical_severity(severity),
+                          false_p=false_positive,
+                          date=date_created,
+                          is_Mitigated=whitehat_sentinel_vuln.get('mitigated', False),
+                          mitigated=mitigated_ts,
+                          last_reviewed=whitehat_sentinel_vuln.get('lastRetested', None),
+                          dynamic_finding=True,
+                          created=date_created,
+                          unique_id_from_tool=whitehat_sentinel_vuln['id'],
+                          url=whitehat_sentinel_vuln.get('')
```

Comment:
url does not appear in the UI. you can ignore it.

Reply:
done!